dbt/adapters/redshift/connections.py

```diff
@@ -104,9 +105,9 @@ def parse(cls, user_sslmode: UserSSLMode) -> "RedshiftSSLConfig":
 @dataclass
 class RedshiftCredentials(Credentials):
     host: str
-    user: str
     port: Port
     method: str = RedshiftConnectionMethod.DATABASE  # type: ignore
+    user: Optional[str] = None  # type: ignore
```

Comment:
if we're making this optional we'll need to add a validation that the user supplied this value for connection methods that require it (i.e. `RedshiftConnectionMethod.DATABASE`)

Reply:
agreed, will work on this and update the code.